src/Uno.UI/Extensions/VisualTreeExtensions.cs

```diff
@@ -0,0 +1,252 @@
+		.OfType<T>()
+		.FirstOrDefault(predicate);
+
+	/// <summary>
+	/// Returns the first descendant of a specified type.
+	/// </summary>
+	/// <typeparam name="T">The type of descendant to find.</typeparam>
+	/// <param name="reference">Any node of the visual tree</param>
+	/// <remarks>The crawling is done depth first.</remarks>
+	public static T? FindFirstDescendant<T>(this _View? reference) => EnumerateDescendants(reference)
+		.OfType<T>()
+		.FirstOrDefault();
+
+	/// <summary>
+	/// Returns the first descendant of a specified type that satisfies the <paramref name="predicate"/>.
+	/// </summary>
+	/// <typeparam name="T">The type of descendant to find.</typeparam>
+	/// <param name="reference">Any node of the visual tree</param>
+	/// <param name="predicate">A function to test each node for a condition.</param>
+	/// <remarks>The crawling is done depth first.</remarks>
+	public static T? FindFirstDescendant<T>(this _View? reference, Func<T, bool> predicate) => EnumerateDescendants(reference)
+		.OfType<T>()
+		.FirstOrDefault(predicate);
+
+	/// <summary>
+	/// Returns the first descendant of a specified type that satisfies the <paramref name="predicate"/> whose ancestors (up to <paramref name="reference"/>) and itself satisfy the <paramref name="hierarchyPredicate"/>.
+	/// </summary>
+	/// <typeparam name="T">The type of descendant to find.</typeparam>
+	/// <param name="reference">Any node of the visual tree</param>
+	/// <param name="hierarchyPredicate">A function to test each ancestor for a condition.</param>
+	/// <param name="predicate">A function to test each descendant for a condition.</param>
+	/// <remarks>The crawling is done depth first.</remarks>
+	public static T? FindFirstDescendant<T>(this _View? reference, Func<_View, bool> hierarchyPredicate, Func<T, bool> predicate) => EnumerateDescendants(reference, hierarchyPredicate)
+		.OfType<T>()
+		.FirstOrDefault(predicate);
+
+	/// <summary>
+	/// Returns all the visual descendants of a node.
+	/// </summary>
+	/// <param name="reference">Any node of the visual tree</param>
+	public static IEnumerable<_View> EnumerateDescendants(this _View? reference) => EnumerateDescendants(reference, x => true);
+
+	/// <summary>
+	/// Returns all the visual descendants of a node that satisfies the <paramref name="hierarchyPredicate"/>.
+	/// </summary>
+	/// <param name="reference">Any node of the visual tree</param>
+	/// <param name="hierarchyPredicate"></param>
+	/// <returns></returns>
+	public static IEnumerable<_View> EnumerateDescendants(this _View? reference, Func<_View, bool> hierarchyPredicate)
+	{
+		foreach (var child in reference.EnumerateChildren().Where(hierarchyPredicate))
+		{
+			yield return child;
+			foreach (var grandchild in child.EnumerateDescendants(hierarchyPredicate))
+			{
+				yield return grandchild;
+			}
+		}
+	}
+}
+
+public static partial class VisualTreeExtensions
```

Comment:
Why not put everything in the same class ?

Reply:
im using different set of signature for `_View`:
platforms|ViewExtensions.cs|VisualTreeExtensions
-|-|-
ios/macos|UIView/NSView|UIView/NSView
android|**ViewGroup**|**View**
others|**UIElement**|**DependencyObject**

Now to think of it, perhaps I can just put the code into `ViewExtensions.visual-tree.cs` or use a different identifier like `_VisualTreeNode`?

Reply:
Yes, lets put those in different files at least. That will help reading it.